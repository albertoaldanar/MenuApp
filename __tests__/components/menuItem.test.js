import { Text } from "react-native";
import Enzyme, { shallow } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";

Enzyme.configure({ adapter: new Adapter() });

import React from "react";
import MenuItem from "../../src/components/menuItem";

let wrapper;

describe("<MenuItem />", () => {
  beforeAll(() => {
    const props = {
      menuItem: {
        description: "Test menu description",
        name: "Test restaurant",
        price: 50,
      },
    };

    wrapper = shallow(<MenuItem {...props} />);
  });
  it("MenuItem component is defined", () => {
    expect(wrapper).toBeDefined();
  });

  it("Menu item renders 3 Text elements correctly. Name, description and price", () => {
    expect(wrapper.find(Text)).toHaveLength(3);
  });
  it("Menu Item snapshot test", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
