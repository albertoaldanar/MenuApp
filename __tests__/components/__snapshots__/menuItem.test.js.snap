// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuItem /> Menu Item snapshot test 1`] = `ShallowWrapper {}`;
